test/mpi/coll/test_coll_algos.sh

```diff
@@ -0,0 +1,31 @@
+nl='
```

Comment:
Needs header and "#!/bin/sh" line. See `test/mpi/autogen.sh` for example. Also, can we just use `\n` directly for newline? I'm not fond of this method here.

Reply:
Yeah, I'll add those. I think there was some reason that @rkalidas didn't use `\n` directly, but I can't remember what it was. I'll try it out and see if I can remember.

Reply:
So I can't find any specific documentation on it, but it seems that `AC_SUBST` doesn't expand `\n` characters in environment variables into newlines.

It seems to do something like `echo $var` instead of `echo -e $var`. The first doesn't expand the `-e` while the second does. So I guess we either need to keep this or figure out some more autoconf magic. I vote for the first one.

Reply:
I did go ahead and fix the `#!/bin/sh` thing though.

Reply:
K, cool. Yes the `AC_SUBST` documentation even says "This value of variable should not contain literal newlines." So I think we are good here.